### PR TITLE
Updates-Scroll-To-Invalid-For-Mhr-Registration

### PR DESCRIPTION
**Pull request by orelbn**
*Issue #:* /bcgov/entity#16877

*Description of changes:*
- Made a separate scroll to invalid function to account for the uniqueness of the review and confirm page.

*Note for reviewers:*
A lot of the spacing changes are from removing an unnecessary div.

To make it easier to review you can hide whitespaces:
image

I don't believe that `@registration-incomplete="registrationIncomplete()"`  is needed since we are intercepting the section that could ultimately lead to that value being emitted. If you need further details please reach out via rocketChat, as it will take me awhile to explain. 

I think, we should clean up the `ButtonFooter` component a bit to remove some unnecessary logic, and make things easier to think through, but ill leave that to future discussions.

*screenshots:*

---
Before PR:

View after scroll to review and confirm components:

Authorization:
Before-Authorization

Staff payments:
before-staff-payment

---

After PR:

View after scroll to review and confirm components:

Authorization:
Authorization

Staff-payment:
StaffPayment

View after scroll to steps:
Your Home:
YourHome

Submitting Party:
SubmittingParty

HomeOwners:
HomeOwners

Location of Home:
Location of Home
---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1412-9t46c7ea.web.app

**Review by cameron-eyds — APPROVED** [review]
Looks good!

File: ppr-ui/src/composables/mhrRegistration/useMhrValidations.ts [hunk 1]
```diff
@@ -44,29 +44,45 @@ export const useMhrValidations = (validationState: any) => {
...
+  }
+
+  /** Scroll to first HEADER of invalid step in the view.
+   *  If all steps are valid scroll to first Section that is invalid
+   *  in the specified section area.
+   */
+  const scrollToInvalidReviewConfirm = (stepsValidation: Array<boolean>): boolean => {
+    // Find the _first_ corresponding step that is invalid in the specified view
+    const view = document.getElementById('mhr-review-confirm')
+    const invalidStep = view.getElementsByTagName('header')[stepsValidation.indexOf(false)]
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrRegistration/useMhrValidations.ts]: Should probably conditionally check the `view`: `view?.getElementsByTagName...`

**Review by dimak1 — APPROVED** [review]
Looks good, approved with a comment.

**Review by doug-lovett — APPROVED** [review]
Very nice again.

File: ppr-ui/src/composables/mhrRegistration/useMhrValidations.ts (hunk 1 above)
Comment (orelbn) [reply ppr-ui/src/composables/mhrRegistration/useMhrValidations.ts]: Good call thanks!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1412-9t46c7ea.web.app

**Pull request merged**